src/NearestNeighbors.jl

```diff
@@ -128,26 +129,38 @@ end
 function MMI.predict(m::KNNRegressor, (tree, y, w), X)
     Xmatrix     = MMI.matrix(X, transpose=true) # NOTE: copies the data
     idxs, dists = NN.knn(tree, Xmatrix, m.K)
-    preds       = zeros(length(idxs))
+
+    if typeof(y) <: AbstractVector
```

Comment:
I would have prefered creating an internal method `_predict_knn_reg(idxs, dists, y::AbstractVector)` to dispatch on.